misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/util/ConverterUtils.java

```diff
@@ -74,6 +87,89 @@ public static String escapeIdentifier(String identifier) {
         }
     }
 
+    /**
+     * This method returns existing Types on a module/file(for single file projects).
+     *
+     * @param workspaceManager Workspace manager instance
+     * @param filePathUri FilePath URI of the/a file in a singleFileProject or module
+     * @return {@link List<String>} List of already existing Types
+     */
+    public static List<String> getExistingTypeNames(WorkspaceManager workspaceManager, String filePathUri) {
+        List<String> existingTypeNames = new ArrayList<>();
+        if (filePathUri == null) {
+            return existingTypeNames;
+        }
+
+        Path filePathResolved;
+        try {
+            URI filePathUriResolved = new URI(filePathUri);
+            filePathResolved = Path.of(filePathUriResolved);
+        } catch (InvalidPathException | URISyntaxException e) {
+            return existingTypeNames;
+        }
+
+        if (workspaceManager != null && workspaceManager.semanticModel(filePathResolved).isPresent()) {
+            List<Symbol> moduleSymbols = workspaceManager.semanticModel(filePathResolved).get().moduleSymbols();
+            moduleSymbols.forEach(symbol -> {
+                if (symbol.getName().isPresent()) {
+                    existingTypeNames.add(symbol.getName().get());
+                }
+            });
+            return existingTypeNames;
+        }
+
+        Project project;
+        // Check if the provided file is a SingleFileProject
+        try {
+            project = SingleFileProject.load(filePathResolved);
+            List<Symbol> moduleSymbols =
+                    project.currentPackage().getDefaultModule().getCompilation().getSemanticModel().moduleSymbols();
+            moduleSymbols.forEach(symbol -> {
+                if (symbol.getName().isPresent()) {
+                    existingTypeNames.add(symbol.getName().get());
+                }
+            });
+        } catch (ProjectException pe) {
+            // Check if the provided file is a part of BuildProject
+            Path projectRoot = ProjectUtils.findProjectRoot(filePathResolved);
+            if (projectRoot != null) {
+                try {
+                    project = BuildProject.load(projectRoot);
+                    List<Symbol> moduleSymbols = project.currentPackage()
+                            .module(project.documentId(filePathResolved).moduleId())
+                            .getCompilation().getSemanticModel().moduleSymbols();
+                    moduleSymbols.forEach(symbol -> {
+                        if (symbol.getName().isPresent()) {
+                            existingTypeNames.add(symbol.getName().get());
+                        }
+                    });
+                } catch (ProjectException pe1) {
+                    return existingTypeNames;
+                }
+            }
```

Comment:
Instead of handling build projects inside the catch block, cant we first check for the project root and, branch out into the two separate logics(single file and package) based on the output?

Reply:
Updated as per the suggestion. Please review and resolve the conversation. @NipunaRanasinghe